### DISCUSSION TITLE
deleting an image after changing it

### DISCUSSION THREAD

**Discussion by ano101** [filamentphp/filament#15034] [Help]
### Package

Form builder

### Package Version

v3

### How can we help you?

Hello everyone. I have a fileupload in json builder, I need to delete the old file on disk when uploading a new one, how can I do this?

**Comment by ano101** [comment]
when uploading an image, I generate a title with the current date

**Reply by leandrocfe** [reply]
_It is the responsibility of the developer to delete these files from the disk if they are removed, as Filament is unaware if they are depended on elsewhere. One way to do this automatically is observing a [model event](https://laravel.com/docs/eloquent#events)._
